### PR TITLE
Feature/psm

### PR DESCRIPTION
**Pull request by RoseKapps**
PSM finished and tested
the i2c need to be physically fixed
just need to add README file

### CODE REVIEW

File: key
Comment (PizzaAllTheWay) [inline key:1]: Delete this file

File: key.pub
Comment (PizzaAllTheWay) [inline key.pub:1]: Delete this file

File: mission/internal_status/internal_status/power_sense_module_lib.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        self.bus = smbus.SMBus(1)
+        self.channel_voltage = MCP342x(self.bus,
+                                       self.i2c_adress,
+                                       channel=1,
+                                       resolution=18)  # voltage
+        self.channel_current = MCP342x(self.bus,
+                                       self.i2c_adress,
+                                       channel=0,
+                                       resolution=18)  # current   
+        #time.sleep(1)
```
Comment (PizzaAllTheWay) [inline mission/internal_status/internal_status/power_sense_module_lib.py:22]: Delete the "time.sleep(1)" coment

File: mission/internal_status/internal_status/power_sense_module_publisher.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+
+    def __init__(self):
+        super().__init__('PSM_publisher')
+        self.PSM = internal_status.power_sense_module_lib.PowerSenseModule()
+        self.publisher_current = self.create_publisher(Float32, '/asv/power_sense_module/current', 1)
+        self.publisher_voltage = self.create_publisher(Float32, '/asv/power_sense_module/voltage', 1)
+        timer_period = 0.5
+        self.timer = self.create_timer(timer_period, self.timer_callback)
+        self.i = 0
```
Comment (PizzaAllTheWay) [inline mission/internal_status/internal_status/power_sense_module_publisher.py:17]: remove unecessary variable self.i as it serves no purpose

File: mission/internal_status/internal_status/power_sense_module_publisher.py [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+        current = Float32()
+        voltage = Float32()
+        #call the two functions of the power_sense_module_lib to get the current and voltage values of the PSM
+        current.data = self.PSM.get_current()
+        voltage.data = self.PSM.get_voltage()
+        self.publisher_current.publish(current)  #publish current value to the "current topic" 
+        self.get_logger().info('Publishing PSM current: "%s"' % current.data)
+        self.publisher_voltage.publish(voltage)  #publish voltage value to the "voltge topic"
+        self.get_logger().info('Publishing PSM voltage: "%s"' % voltage.data)
+        self.i += 1
```
Comment (PizzaAllTheWay) [inline mission/internal_status/internal_status/power_sense_module_publisher.py:29]: remove unecessary variable self.i as it serves no purpose

File: mission/internal_status/test/test_copyright.py
Comment (PizzaAllTheWay) [inline mission/internal_status/test/test_copyright.py:1]: Do we need this file?

File: mission/internal_status/test/test_flake8.py
Comment (PizzaAllTheWay) [inline mission/internal_status/test/test_flake8.py:1]: Do we need this file?

File: mission/internal_status/test/test_pep257.py
Comment (PizzaAllTheWay) [inline mission/internal_status/test/test_pep257.py:1]: Do we need this file?

**Review by PizzaAllTheWay — CHANGES_REQUESTED** [review]
Overal very good job Rose. Just need to dlete some stuff and add a README.md file and you are good ^-^

File: mission/internal_status/test/test_pep257.py
Comment (chrstrom) [reply mission/internal_status/test/test_pep257.py:1]: Might want to handover to devops, to make flake8, pep257 and mypy checks run on all files across this repo 😉

File: mission/internal_status/internal_status/power_sense_module_lib.py [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    def get_current(self):
+        try:
+            system_current = (self.channel_current.convert_and_read() - 
+                            self.psm_to_battery_current_offset
+                            ) * self.psm_to_battery_current_scale_factor
+
+            return system_current
+
+        except IOError:
+            return
```
Comment (chrstrom) [inline mission/internal_status/internal_status/power_sense_module_lib.py:49]: You could avoid a bit of copy paste / nesting here by adding a decorator, f.ex:

```python
def io_error_safe_read(func):
    def wrapper(*args, **kwargs):
        try:
            return func(*args, **kwargs)
        except IOError:
            return
    return wrapper
```

then:

```python
    @io_error_safe_read
    def get_voltage(self):
        system_voltage = (self.channel_voltage.convert_and_read() *
                          self.psm_to_battery_voltage)
        return system_voltage

    @io_error_safe_read
    def get_current(self):
        system_current = (self.channel_current.convert_and_read() - 
                          self.psm_to_battery_current_offset
                         ) * self.psm_to_battery_current_scale_factor
        return system_current
```

File: mission/internal_status/internal_status/power_sense_module_publisher.py [hunk 5]
```diff
@@ -0,0 +1,49 @@
+import internal_status.power_sense_module_lib
+import rclpy
+from rclpy.node import Node
+from std_msgs.msg import Float32
+
+
+class MinimalPublisher(Node):
```
Comment (chrstrom) [inline mission/internal_status/internal_status/power_sense_module_publisher.py:7]: Would rename this to something a bit more descriptive 🐱

File: mission/internal_status/test/test_pep257.py
Comment (PizzaAllTheWay) [reply mission/internal_status/test/test_pep257.py:1]: @FabianKongelf DevOps project O-O <3

File: key
Comment (PizzaAllTheWay) [inline key:1]: @RoseKapps Delete this file

File: key.pub
Comment (PizzaAllTheWay) [inline key.pub:1]: @RoseKapps Delete this file

**Review by PizzaAllTheWay — CHANGES_REQUESTED** [review]
Verry well done. 
Verry IMPORTANT that you delete the "key" and "key.gen" files as I can connect to you git acount and push code wothout your concent and without knowing your password. I sugest you also disable this key just in case and make one private and write it down somewhere :)

If you want to further improve PSM system you can try and implement what Cristopher has commented. But that is ONLY if you feel confident that you know how to do those changes, otherwise no further improvements needed, this is a good code for me :)

**Review by PizzaAllTheWay — APPROVED** [review]
Verry nice jesjes

**Pull request merged**